### PR TITLE
merge #13269, #13425,  #13557,  #13721,  #13666, #13723: BIP 174 PSBT Serializations and RPCs

### PR DESCRIPTION
**Pull request by kittywhiskers**
# Overview

This pull request aims to implement [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki) ("Partially Signed Bitcoin Transaction Format" by Andrew Chow) *without* references to segregated witness (which featured heavily in the implementation), a leftover reference unimplemented Replace-by-fee (RBF) transactions and relevant changes made to CoinJoin logic that accompany the PR

## Motivation

During initial preparation for https://github.com/bitcoin/bitcoin/pull/18242 ("Add BIP324 encrypted p2p transport de-/serializer" by jonasschnelli) and the backporting of https://github.com/bitcoin/bitcoin/commit/78c312c983255e15fc274de2368a2ec13ce81cbf, a difference in class and function names were noticed and that in turn lead to a series of pull requests the lead us to this one.

The motivation is to reduce future merge conflicts when backporting pull requests from Bitcoin Core.

## Contents

#13269, #13425,  #13557,  #13721,  #13666, #13723

# Disclosures

* **This is a work in progress and needs external review**. Dash-specific changes have not been tested, only compilation and unit test success has been ensured (P.S.: It fails for now). Running the client on a testnet _is_ necessary.
* Tests fail. I do not know why, I do not know how. They fail. (see "Changes that need review")

## Changes that need review

* Anything and everything related to CoinJoin

### CODE REVIEW

**Comment by UdjinM6** [comment]
pls see https://github.com/UdjinM6/dash/commits/pr4186 for some patches (besides 52f38ff1d0112b70c7759cb6ae35f7755c9ae6d0 which is a temporary hack to "fix" tests)

**Comment by PastaPastaPasta** [comment]
Need to rebase to drop https://github.com/dashpay/dash/pull/4186/commits/0297bfb1d7074f7e4f92f2cc4db700a1a7c7fd68

**Comment by kittywhiskers** [comment]
Rebased and updated, Python tests pending

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
dae890b516361c089d1181989d57a0ff602a5732 and b60872b89e7d5eddfa36ce71a9f9ee338a8bef4e should be dropped (13269 and 13425 were fully backported earlier + changes there do not match original PRs, probably some leftovers from rebase), plus see below

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -5624,3 +5624,16 @@ std::vector<OutputGroup> CWallet::GroupOutputs(const std::vector<COutput>& outpu
...
+    {
+        LOCK(cs_wallet);
+        auto it = mapKeyMetadata.find(keyID);
+        if (it != mapKeyMetadata.end()) {
+            meta = it->second;
+        }
+    }
+    std::copy(keyID.begin(), keyID.begin() + 4, info.fingerprint);
+    return true;
+}
```
Comment (PastaPastaPasta) [inline src/wallet/wallet.cpp:5639]: nit

**Review by PastaPastaPasta — APPROVED** [review]
tests passed, looks good to me. Let's get this ****ing merged!

**Pull request merged**